### PR TITLE
Drop, rather than clear, removed blueprint entities

### PR DESCRIPTION
**Pull request by jleibs**
### What
- Resolves: https://github.com/rerun-io/rerun/issues/6552

This makes it so we no longer care about fancy logic to drop cleared entities with respect to query-time-clear behavior.

Observation. It would be also be quite fast for someone to add a "Drop this entity" context menu :thinking: 

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using examples from latest `main` build: [rerun.io/viewer](https://rerun.io/viewer/pr/7120?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [rerun.io/viewer](https://rerun.io/viewer/pr/7120?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](https://github.com/rerun-io/rerun/blob/main/tests/python/release_checklist)!
* [x] If have noted any breaking changes to the log API in `CHANGELOG.md` and the migration guide

- [PR Build Summary](https://build.rerun.io/pr/7120)
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

To run all checks from `main`, comment on the PR with `@rerun-bot full-check`.

### CODE REVIEW

**Comment by jleibs** [comment]
From, @teh-cmc :
> One issue, which predates this PR, is that ChunkStore::drop_entity_path doesn't seem to notify its subscribers of the changes. Ill fix that tomorrow.
> The other issue, more related to this PR, is that this bypasses EntityDb and therefore EntityDb will not notify its subscribers of the changes. We should introduce a higher-level EntityDb::drop_entity_path that properly forwards the resulting ChunkStoreEvents to its internal subscribers.

File: crates/store/re_entity_db/src/entity_db.rs [hunk 1]
```diff
@@ -205,6 +205,11 @@ impl EntityDb {
         &self.data_store
     }
 
+    #[inline]
+    pub fn store_mut(&mut self) -> &mut ChunkStore {
+        &mut self.data_store
+    }
```
Comment (emilk) [inline crates/store/re_entity_db/src/entity_db.rs:211]: I rather we did not do this, as it opens up the possibility of bugs like calling `entity_db.store_mut().insert_chunk(…)`.

See also:

https://github.com/rerun-io/rerun/blob/c19457d18bb48fca07f3c603a4e077e7e3598417/crates/store/re_entity_db/src/entity_db.rs#L27-L28

File: crates/viewer/re_viewer/src/app.rs [hunk 2]
```diff
@@ -517,6 +517,22 @@ impl App {
...
+
+                if let Some(tree) = blueprint_db.tree().subtree(&entity_path) {
+                    tree.visit_children_recursively(|path| {
+                        to_drop.push(path.clone());
+                    });
+                }
+
+                for path in to_drop {
+                    blueprint_db.store_mut().drop_entity_path(&path);
+                }
```
Comment (emilk) [inline crates/viewer/re_viewer/src/app.rs:534]:
```suggestion
                blueprint_db.drop_entity(&entity_path);
```

**Review by emilk — APPROVED** [review]
Looks good, except for exposing `&mut ChunkStore`

**Pull request merged**